app/src/main/java/com/hedvig/app/feature/offer/usecase/GetBundleVariantUseCase.kt

```diff
@@ -0,0 +1,34 @@
+package com.hedvig.app.feature.offer.usecase
+
+import arrow.core.Either
+import com.hedvig.app.feature.offer.OfferRepository
+import com.hedvig.app.feature.offer.model.OfferModel
+import com.hedvig.app.feature.offer.model.QuoteBundleVariant
+import com.hedvig.app.util.ErrorMessage
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.combine
+
+class GetBundleVariantUseCase(
```

Comment:
Not super comfortable having a UseCase hold state like this instead of being a pure function with simple consistent parameters in/ results out concept. Especially in this case where we're using this UseCase inside `CheckoutViewModel` only to observe the result, meaning that we're relying on the same usecase to be provided to that and to `OfferViewModel` which it does happen in our case since we're providing it with a koin `single` but still not sure if we want to rely on such behaviors with UseCases.
I could be wrong too though :D

Reply:
What do you suggest? Move to OfferViewModel?

Reply:
Hmm I thought about it a bit, and since this is used from the CheckoutViewModel it wouldn't work.
If anything, it'd have to be raised one layer higher, potentially saved in DataStore or something like that which would be a reliable storage.
However, what that would do is have the same practical result as this StateFlow here, which is like a form of storage, just in-memory. I'll make a PR with a suggestion, but this shouldn't block this merge imo.

Reply:
I dont think datastore is the right use case for this tbh. I feel datastore should be app-wide storage where as this is bound to the offer screen lifecycle only.

Reply:
Right, but we're using this inside `CheckoutViewModel` as well, which escapes that screens scope, doesn't it?

Reply:
True, its still bound to the "checkout-lifecycle" though. Datastore should be for data we want to keep between app lifecycles. We would in this case also need to "reset" this selected variant to null or empty after we leave the offer flow.

Reply:
Yeah, with the current impl it will also persist since the UseCase is a single one persisted in the app as well, even if you leave this flow. But I guess it will reset if the app is killed, which wouldn't happen with a more persistent storage.
To avoid wasting time on this I think we should just merge it and see if we need to do any changes to it in the future.

Reply:
I renamed this use case, hopefully it makes more sense